src/usr/bin/google_set_hostname

```diff
@@ -16,18 +16,17 @@
 # Deal with a new hostname assignment.
 
 if [ -n "$new_host_name" ] && [ -n "$new_ip_address" ]; then
-  # Don't allow DHCP responses with the MDS as the hostname.
-  # See: https://github.com/irsl/gcp-dhcp-takeover-code-exec
-  if [[ "$new_host_name" =~ "metadata.google.internal" ]]; then
-    echo 'not setting invalid hostname'
-    exit 0
-  fi
-
   # Delete entries with new_host_name or new_ip_address in /etc/hosts.
   sed -i"" '/Added by Google/d' /etc/hosts
 
-  # Add an entry for our new_host_name/new_ip_address in /etc/hosts.
-  echo "${new_ip_address} ${new_host_name} ${new_host_name%%.*}  # Added by Google" >> /etc/hosts
+  # Don't allow DHCP responses with the MDS as the hostname.
+  # See: https://github.com/irsl/gcp-dhcp-takeover-code-exec
+  if echo "$new_host_name" | grep -iq "metadata.google.internal"; then
```

Comment:
If I'm reading this correctly, this code won't match "metadata" (or "metadata.internal") which is another example of an invalid hostname.

Reply:
yes we only want to match metadata.google.internal